src/libOpenImageIO/maketexture.cpp

```diff
@@ -428,7 +428,110 @@ lightprobe_to_envlatl (ImageBuf &dst, const ImageBuf &src, bool y_is_up,
     return true;
 }
 
+// compute image gradient in s,t space using a Sobel filter
+static void 
+sobel_gradient(const ImageBuf &src, int x, int y, int width, int height, float *h, float* dh_ds, float *dh_dt ){
+            
+    ASSERT (src.spec().nchannels==1);
 
+    // 3x3 pixels block
+    const float* pix[3];
+
+    pix[1] = static_cast<const float*>(src.pixeladdr(x, y));
```

Comment:
Aha, so the problem here is that pixeladdr will return nullptr if the image pixels aren't in memory. It's a little bit of a dangerous method.

In the calling function, you are already using ImageBuf::Iterator to loop over the destination (output) image. I think that here, you should similarly be using ImageBuf::ConstIterator to access the source image. Not only will that be robust to use of the underlying ImageCache, but also you can make the iterators handle the border conditions for you, eliminating all the special cases you have below.

Reply:
So what I'd recommend is that you pass a reference to the dst iterator here, let's call it `ImageBuf::Iterator<float> &dstpix`, and then iterate over the 3x3 region of source like this:

    ImageBuf::ConstIterator<float> srcpix (src, dstpix.x()-1, dstpix.x()+2, dstpix.y()-1, dstpix.y()+2, 0, 1, WrapClamp);
    static const float sobelweight[9] = { ... };
    for (int i = 0; !srcpix.done(); ++srcpix, ++i) {
        // accumulate to dh_ds, etc., using a table weight[] with the 3x3 coefficients
        *dh_ds += sobelweight[i] * src[0];  // handles wrap!
        ...
    }